delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -0,0 +1,510 @@
+        """ Create a ns0:Customer as a dict from picking
+
+        This is the Postlogistic Customer, thus the sender
+
+        :param picking: picking browse record
+        :return a dict containing data for ns0:Customer
+
+        """
+        company = picking.company_id
+        partner = company.partner_id
+
+        customer = {
+            "name1": partner.name,
+            "street": partner.street,
+            "zip": partner.zip,
+            "city": partner.city,
+            "country": partner.country_id.code,
+            "domicilePostOffice": picking.carrier_id.postlogistics_office or None,
+        }
+        logo = picking.carrier_id.postlogistics_logo
+        if logo:
+            logo_image = Image.open(BytesIO(base64.b64decode(logo)))
+            logo_format = logo_image.format
+            customer["logo"] = logo.decode()
+            customer["logoFormat"] = logo_format
+        return customer
+
+    def _get_label_layout(self, picking):
+        """
+        Get Label layout define in carrier
+        """
+        return picking.carrier_id.postlogistics_label_layout.code
+
+    def _get_output_format(self, picking):
+        """
+        Get Output format define in carrier
+        """
+        return picking.carrier_id.postlogistics_output_format.code
+
+    def _get_image_resolution(self, picking):
+        """
+        Get Output Resolution Code define in carrier
+        """
+        return picking.carrier_id.postlogistics_resolution.code
+
+    def _get_license(self, picking):
+        """ Get the license
+
+        Take it from carrier and if not defined get the first license.
+
+        :return: license number
+        """
+        franking_license = picking.carrier_id.postlogistics_license_id
+        return franking_license.number
+
+    def _prepare_attributes(self, picking, pack=None, pack_num=None, pack_total=None):
+        packaging = (
+            pack
+            and pack.packaging_id
+            or picking.carrier_id.postlogistics_default_packaging_id
+        )
+        services = packaging._get_packaging_codes()
+
+        total_weight = pack.shipping_weight if pack else picking.shipping_weight
+        total_weight *= 1000
+
+        if not services:
+            raise exceptions.UserError(
+                _("No Postlogistics packaging services found in this picking.")
+            )
+
+        # Activate phone notification ZAW3213
+        # if phone call notification is set on partner
+        if picking.partner_id.postlogistics_notification == "phone":
+            services.append("ZAW3213")
+
+        attributes = {
+            "weight": int(total_weight),
+        }
+
+        # Remove the services if the delivery fixed date is not set
+        if "ZAW3217" in services:
+            if picking.delivery_fixed_date:
+                attributes["deliveryDate"] = picking.delivery_fixed_date
+            else:
+                services.remove("ZAW3217")
+
+        if pack_num:
```

Comment:
After clarification, parcelNo/Total cannot be used if service is not activated
```
if "ZAW3218" in services:
  if pack_total > 1:
    ...
  else:
     services.remove("ZAW3218")
```

Reply:
Added in this commit https://github.com/OCA/delivery-carrier/pull/323/commits/e17d7aa83b3bcb7d5be1b3cc4a92da0e27c2bf04